internal/resources/aws/elastic_beanstalk_environment.go

```diff
@@ -0,0 +1,103 @@
+//
+// <TODO: Add any important information about the resource and links to the
+// pricing pages or documentation that might be useful to developers in the future, e.g:>
+//
+// Resource information: https://aws.amazon.com/elasticbeanstalk/
+// Pricing information: https://aws.amazon.com/elasticbeanstalk/pricing/
+type ElasticBeanstalkEnvironment struct {
+	Address string
+	Region  string
+	Name    string
+
+	InstanceCount    *int64
+	InstanceType     string
+	RDSIncluded      bool
+	LoadBalancerType string
+
+	StreamLogs            bool
+	MonthlyDataIngestedGB *float64 `infracost_usage:"cloudwatch_monthly_data_ingested_gb"`
+	StorageGB             *float64 `infracost_usage:"cloudwatch_storage_gb"`
+	MonthlyDataScannedGB  *float64 `infracost_usage:"cloudwatch_monthly_data_scanned_gb"`
+	MonthlyDataProcessed  *float64 `infracost_usage:"monthly_data_processed_gb"`
+	LBNewConnections      *float64 `infracost_usage:"lb_new_connections"`
+	LBActiveConnections   *float64 `infracost_usage:"lb_active_connections"`
+	LBRuleEvaluations     *float64 `infracost_usage:"lb_rule_evaluations"`
+	LBProcessedBytesGB    *float64 `infracost_usage:"lb_processed_bytes_gb"`
+
+	RootBlockDevice     *EBSVolume
+	CloudwatchLogGroup  *CloudwatchLogGroup
+	LoadBalancer        *LB
+	ElasticLoadBalancer *ELB
+	DBInstance          *DBInstance
+	LaunchConfiguration *LaunchConfiguration
+}
+
+// ElasticBeanstalkEnvironmentUsageSchema defines a list which represents the usage schema of ElasticBeanstalkEnvironment.
+var ElasticBeanstalkEnvironmentUsageSchema = append([]*schema.UsageItem{
+	{Key: "monthly_data_processed_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "lb_new_connections", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "lb_active_connections", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "lb_rule_evaluations", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "lb_processed_bytes_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "cloudwatch_monthly_data_ingested_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "cloudwatch_monthly_data_scanned_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "cloudwatch_storage_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "instances", DefaultValue: 0, ValueType: schema.Int64},
+}, LaunchConfigurationUsageSchema...)
+
+// PopulateUsage parses the u schema.UsageData into the ElasticBeanstalkEnvironment.
+// It uses the `infracost_usage` struct tags to populate data into the ElasticBeanstalkEnvironment.
+func (r *ElasticBeanstalkEnvironment) PopulateUsage(u *schema.UsageData) {
+	if r.LoadBalancerType == "classic" {
+		resources.PopulateArgsWithUsage(r.ElasticLoadBalancer, u)
+	} else {
+		resources.PopulateArgsWithUsage(r.LoadBalancer, u)
+	}
+	if r.StreamLogs {
+		resources.PopulateArgsWithUsage(r.CloudwatchLogGroup, u)
+	}
+	if r.RDSIncluded {
+		resources.PopulateArgsWithUsage(r.DBInstance, u)
+	}
```

Comment:
So the reason this isn't working with the sub resources is that the sub resources have different usage params from the ones defined here. e.g.  `CloudWatchLogGroup` has a usage param called `monthly_data_ingested_gb` not `cloudwatch_monthly_data_ingested_gb` therefore the sub resource can't parse the usage.

In order to sort i'd introduce a sub object in the usage file. See my comment on `infracost-usage-example` for more info. 

In order to do this need to change the usage schema to something like this:

```yaml
  aws_elastic_beanstalk_environment.my_eb_environment:
    cloudwatch:
      monthly_data_ingested_gb: 50  # Cloudwatch log stream ingest in GB
      monthly_data_scanned_gb: 50   # Cloudwatch data scanned in GB
      storage_gb: 100               # Cloudwatch storage used in GB
```

Then the `ElasticBeanstalkEnvironmentUsageSchema` would change to something like so:

```go
var ElasticBeanstalkEnvironmentUsageSchema = []*schema.UsageItem{
	{
	    Key: "cloudwatch", 
            DefaultValue: CloudwatchLogGroupUsageSchema,
            ValueType: schema.SubResourceUsage
         }
}
```

Then in populate usage you would have to do something like:


```go
func (r *ElasticBeanstalkEnvironment) PopulateUsage(u *schema.UsageData) {
	if r.CloudwatchLogGroup != nil {
		resources.PopulateArgsWithUsage(r.CloudwatchLogGroup, u["cloudwatch"])
	}
}
```